open_spiel/python/mfg/examples/mfg_Proximal_policy_optimization_pytorch.py

```diff
@@ -0,0 +1,448 @@
+    parser.add_argument("--game-setting", type=str, default="crowd_modelling_2d_four_rooms", help="Set the game to benchmark options:(crowd_modelling_2d_four_rooms) and (crowd_modelling_2d_maze)")
+    
+    parser.add_argument("--lr", type=float, default=1e-3, help="Learning rate of the optimizer")
+    parser.add_argument("--num-episodes", type=int, default=5, help="set the number of episodes of to collect per rollout")
+    parser.add_argument("--update-episodes", type=int, default=20, help="set the number of episodes of the inner loop")
+    parser.add_argument("--update-iterations", type=int, default=100, help="Set the number of global update steps of the outer loop")
+    
+    parser.add_argument('--optimizer', type=str, default="Adam", help="Set the optimizer (Adam) or (SGD)")
+    parser.add_argument('--torch-deterministic', type=lambda x:bool(strtobool(x)), default=True, nargs="?", const=True, help="Use to repreduce experiment results")
+    parser.add_argument('--cuda', type=lambda x:bool(strtobool(x)), default=False, nargs='?', const=True, help="Use Gpu to run the experiment")
+
+    
+    # PPO parameters
+    parser.add_argument('--gamma', type=float, default=0.9, help='set discount factor gamma')
+    parser.add_argument("--num-minibatches", type=int, default=5,  help="the number of mini-batches")
+    parser.add_argument("--update-epochs", type=int, default=5, help="the K epochs to update the policy")
+    parser.add_argument("--clip-coef", type=float, default=0.2, help="the surrogate clipping coefficient")
+    parser.add_argument("--ent-coef", type=float, default=0.01, help="coefficient of the entropy")
+    parser.add_argument("--max-grad-norm", type=float, default=5, help="the maximum norm for the gradient clipping")
+
+
+    # MFPPO parameters
+    parser.add_argument('--alpha', type= int, default=0.5, help='Set alpha to controll the iteration and epsiode policy updates')
+    parser.add_argument('--eps-eps', type= int, default=0.2, help='eps to update the episode learned policy')
+    parser.add_argument('--itr-eps', type= int, default=0.05, help='eps to update the episode learned policy')
+
+    args = parser.parse_args()
+
+    return args
+
+
+class NashC(NashConv):
+    # Mainly used to calculate the exploitability 
+    def __init__(self, game,distrib,pi_value, root_state=None):
+        self._game = game
+        if root_state is None:
+            self._root_states = game.new_initial_states()
+        else:
+            self._root_states = [root_state]
+            
+        self._distrib = distrib
+
+        self._pi_value = pi_value
+
+        self._br_value = best_response_value.BestResponse(
+            self._game,
+            self._distrib,
+            value.TabularValueFunction(self._game),
+            root_state=root_state)
+
+
+class Agent(nn.Module):
+    def __init__(self, info_state_size, num_actions):
```

Comment:
2-space indent, as the rest of the python code in OpenSpiel

Reply:
Indentation changed from 4 spaces to 2 spaces, I did it in the next commit  ca0e22c